### PR TITLE
Feature/segregate core tests

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

File: .github/workflows/overall-tests.yml [hunk 1]
```diff
@@ -41,7 +50,22 @@ jobs:
         run: pipenv run playwright install chromium --with-deps
 
       - name: Pytest
-        run: pipenv run pytest --cov=taipy --cov-append --cov-report="xml:overall-coverage.xml" --cov-report term-missing tests
+        run: pipenv run pytest -m "not orchestrator_dispatcher and not modin and not standalone" --cov=taipy --cov-append --cov-report="xml:overall-coverage.xml" --cov-report term-missing tests
+
+      - name: Pytest Core Submit
+        if: steps.changes.outputs.core == 'true'
```
Comment (jrobinAV) [inline .github/workflows/overall-tests.yml]:
```suggestion
```

File: .github/workflows/overall-tests.yml [hunk 2]
```diff
@@ -41,7 +50,22 @@ jobs:
...
+        run: |
+          pipenv run pytest -m "orchestrator_dispatcher" tests/core
+
+      - name: Pytest Core Standalone
+        if: steps.changes.outputs.core == 'true'
```
Comment (jrobinAV) [inline .github/workflows/overall-tests.yml]:
```suggestion
```

File: .github/workflows/overall-tests.yml [hunk 3]
```diff
@@ -41,7 +50,22 @@ jobs:
...
+        run: |
+          pipenv run pytest -m "orchestrator_dispatcher" tests/core
+
+      - name: Pytest Core Modin
+        if: steps.changes.outputs.core == 'true'
```
Comment (jrobinAV) [inline .github/workflows/overall-tests.yml]:
```suggestion
```

File: taipy/core/_core.py [hunk 4]
```diff
@@ -51,6 +51,7 @@ def run(self, force_restart=False):
         This function checks the configuration, manages application's version,
         and starts a dispatcher and lock the Config.
         """
+        # Just a comment to trigger the tests.
```
Comment (jrobinAV) [inline taipy/core/_core.py]: Nice comment 🤣

**Review by jrobinAV — DISMISSED** [review]

**Review by florian-vuillemot — APPROVED** [review]
Well done!
Is it time to use cache to speed up the configuration process on the action? Or perhaps create an artifact?

**Review by jrobinAV — APPROVED** [review]

**Comment by joaoandre-avaiga** [comment]
> Well done! Is it time to use cache to speed up the configuration process on the action? Or perhaps create an artifact?

I think so. Next step would be caching dependencies. Most of all we need to profile our unit tests and see whats going on, they are somewhat slow. Caching can help but I think a small refactor is needed as well.

**Pull request merged**